src/widgets/colorcircle.cpp

```diff
@@ -0,0 +1,203 @@
+#include "colorcircle.h"
+
+#include <QPainter>
+#include <QVECTOR2D>
+#include <qmath.h>
+#include <QMOUSEEVENT>
+#include <QSTYLEOPTION>
+
+ColorCircle::ColorCircle(QWidget *parent, QColor ic) : QWidget(parent)
+{
+	setGeometry(4, 4, parent->width() - 4, parent->height() - 4);
+	radius = width() / 2 - 4;
+	centerPoint.setX(radius + 4);
+	centerPoint.setY(radius + 4);
```

Comment:
Too many magic numbers, what are these? What if I want to change them later on? Make these configurable vars.